content/articles/leveraging-gptneo-to-generate-ai--based-blog-content/index.md

```diff
@@ -0,0 +1,150 @@
+
+Let's go ahead and install it into our notebook.
+
+```bash
+!pip install transformers
+```
+Now that we've installed them, we need to import them.
+
+```python
+from transformers import pipeline
+```
+This command imports all the available pipelines inside the transformers library. If you want a deeper dive into all the available pipelines, please read this [documentation](https://huggingface.co/docs/transformers/index).
+
+#### Setting up the generator
+This step involves setting up the generator to allow us to generate blog content. 
+
+```python
+generator = pipeline('text-generation', model ='EleutherAI/gpt-neo-2.7B')
+```
+This block of code goes ahead and downloads our GPT-Neo model with 2.7 billion parameters from transformers. Specifically, focusing on the `text-generation` pipeline. We store this model inside a variable known as `generator`. This process is going to take a bit of time to download as it is a fairly large model, about 10GB. 
+
+There are the 1.3 billion and 125 million parameter models, if your computer is running slow, you can leverage that model instead using the following command:
+
+For the 1.3 billion parameter model, use:
+
+```python
+generator = pipeline('text-generation', model='EleutherAI/gpt-neo-1.3B')
+```
+For the 125 million parameter model, use:
+
+```python
+generator = pipeline('text-generation', model ='EleutherAI/gpt-neo-125M')
+```
+
+#### Generating text 
+
+```python
+post = "This is my first blog post, I'm really excited!"
+```
+We need to pass this `post` to our generator to generate posts. We do this by writing the following command:
+
+```python
+result = generator(post, max_length=50, do_sample=True, temperature=0.9)
+```
+After running the above block of code, we'll have our generated text stored inside the `result` variable.
+
+We've passed in some arguments inside the `generator` method. They include:
+
+- `post` is the string we created earlier. It contains the text we want the model to generate. 
+- `max_length` defines how long your output is going to be. For our case, it's 50 words. 
+- `do_sample` is set to `true` to allow for sampling within our model.
+- The `temperature` is set to 0.9. It is the value used to model the next set of probabilities. 
+
+You can play around with these parameters, and tune them if you so wish.
+
+If we take a look at the results, we will see that it has generated a new block of text. 
+
+```bash
+[{'generated_text': "This is my first blog post, I'm really excited! I have been writing a lot lately, and I'm not getting around to it yet. Here's some of what I'm reading in my mind so I can get back to it.\n"}]
+```
+If you'd like to generate more text, you need to increase the maximum length. 
+
+Let's now integrate this model into the Gradio app.
+
+### Integrating the model into the Gradio app
```

Comment:
A keen reader may wonder, since you've already received output from the blog generator at this point, what is the purpose of the subsequent section of integrating into Gradio?

Reply:
Basically, it is meant to make it more user friendly. That's it's sole purpose. If a reader would want to build one for production, that would be the way to go.